model_bakery/baker.py

```diff
@@ -367,6 +367,10 @@ def _make(
                         "{0} iterator is empty.".format(field.name)
                     )
 
+        for k, v in self.model_attrs.items():
+            if is_iterator(v):
+                self.model_attrs[k] = next(v)
```

Comment:
This is not ideal because it's bypassing all of bakery's default behavior. The proper way to do so is to make sure that `self.iterator_attrs` has the iterator object and the `model_attrs` doesn't. 

From what I saw, probably the issue is under the `self.get_fields` methods because it's not listing the related name as a possible field for the `User` model.